test/node-fetch/main.js

```diff
@@ -882,7 +882,7 @@ describe('node-fetch', () => {
           return expect(res.text())
             .to.eventually.be.rejected
             .and.be.an.instanceof(Error)
-            .and.have.property('name', 'AbortError')
+            .and.have.property('name', 'TypeError')
```

Comment:
@dominic I'm actually unsure what the spec expects here. Would you mind help clarifying?

Reply:
It should be `AbortError`.

https://fetch.spec.whatwg.org/#fetch-method -> https://fetch.spec.whatwg.org/#abort-fetch

Reply:
> If response’s body is not null and is readable, then error response’s body with error.

This does not say how that error should propagate through e.g. `.text()`

Reply:
erroring a stream makes it disturbed and the reading algorithm says to throw a TypeError if disturbed.

Reply:
@szmarczak I think this is the relevant part: https://fetch.spec.whatwg.org/#concept-body-consume-body

Reply:
Remember it's not fetch that throws here, we already have resolved the promise, it's the body mixing methods that throw here. wdyt?

Reply:
Makes sense, I guess you're right.

```js
const start = Date.now();
const controller = new AbortController();
const response = await fetch('https://speed.hetzner.de/100MB.bin', {signal: controller.signal});
console.log(Date.now() - start, response.bodyUsed);
(async () => {
    try {
        await response.text();
        console.log(Date.now() - start, 'read');
    } catch (error) {
        console.error(Date.now() - start, error);
    }
})();
console.log(Date.now() - start, 'aborting');
controller.abort();
```

Google Chrome: `DOMException: The user aborted a request.`
Firefox: `DOMException: The operation was aborted.`